editor/modes/visual-editor/block-html.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress Dependencies
+ */
+import { isEqual } from 'lodash';
+import { Component } from '@wordpress/element';
+import { getBlockContent, getSourcedAttributes, getBlockType, isValidBlock } from '@wordpress/blocks';
+
+/**
+ * External Dependencies
+ */
+import { connect } from 'react-redux';
+import TextareaAutosize from 'react-autosize-textarea';
+
+/**
+ * Internal Dependencies
+ */
+import { updateBlock } from '../../actions';
+import { getBlock } from '../../selectors';
+
+class BlockHTML extends Component {
+	constructor( props ) {
+		super( ...arguments );
+		this.onChange = this.onChange.bind( this );
+		this.onBlur = this.onBlur.bind( this );
+		this.state = {
+			html: props.block.isValid ? getBlockContent( props.block ) : props.block.originalContent,
+			attributes: props.block.attributes,
```

Comment:
Why do we need to maintain `attributes` in state?

Reply:
The idea is:

 - don't save the updated HTML (which means parse the attributes) on every key stroke (weird things happen if you do so :) without mentioning the performance reasons) and only persist these changes onBlur
 - If the attributes change externally, this means something else changed the current block,  will have to recompute the `HTML` again.
 - Each time the text mode changes the attributes we store them in state to compare with the new prop.

If you have another suggestion to achieve this, I'm all for it.

Reply:
>don't save the updated HTML (which means parse the attributes) on every key stroke (weird things happen if you do so :) without mentioning the performance reasons) and only persist these changes onBlur

This is fine, I understand we don't want to update the block until blur, so storing initial HTML makes sense. Doesn't seem to imply we need `attributes` to do so, though.

>If the attributes change externally, this means something else changed the current block, will have to recompute the HTML again.

>Each time the text mode changes the attributes we store them in state to compare with the new prop.

We can compare `isEqual( this.props.attributes, nextProps.block.attributes )` in `componentWillReceiveProps`

Reply:
you're right :+1: